### PR TITLE
wifi: Extend AP events

### PR DESCRIPTION
**Pull request by krish2718**
Depends on https://github.com/zephyrproject-rtos/zephyr/pull/66876

### CODE REVIEW

File: subsys/net/l2/wifi/wifi_shell.c [hunk 1]
```diff
@@ -302,6 +308,87 @@ static void handle_wifi_twt_event(struct net_mgmt_event_callback *cb)
...
+	      net_sprint_ll_addr_buf(sta_info->mac, WIFI_MAC_ADDR_LEN,
+				     mac_string_buf, sizeof(mac_string_buf)));
+	print(context.sh, SHELL_NORMAL, "====================================\n");
+	print(context.sh, SHELL_NORMAL, "Link mode: %s\n",
+	      wifi_link_mode_txt(sta_info->link_mode));
+	print(context.sh, SHELL_NORMAL, "TWT: %s\n",
+	      sta_info->twt_capable ? "Supported" : "Not supported");
+
+	k_mutex_lock(&wifi_ap_sta_list_lock, K_FOREVER);
+	sta_info_tmp = k_malloc(sizeof(struct wifi_ap_sta_info));
```
Comment (krish2718) [inline subsys/net/l2/wifi/wifi_shell.c:362]: As I had guessed, this isn't ideal as some platforms don't have the HEAP. So, will remove the HEAP usage.

File: include/zephyr/net/wifi_mgmt.h [hunk 2]
```diff
@@ -568,6 +578,18 @@ struct wifi_raw_scan_result {
...
+/** AP mode - connected STA details */
+struct wifi_ap_sta_info {
+	/** MAC address */
+	uint8_t mac[WIFI_MAC_ADDR_LEN];
+	/** MAC address length */
+	uint8_t mac_length;
+	/** Link mode, see enum wifi_link_mode */
+	enum wifi_link_mode link_mode;
+	/** is TWT capable ? */
+	bool twt_capable;
```
Comment (jukkar) [inline include/zephyr/net/wifi_mgmt.h:590]: Minor nit, but we could save 4 bytes memory by ordering the fields to
```
	/** Link mode, see enum wifi_link_mode */
	enum wifi_link_mode link_mode;
	/** MAC address */
	uint8_t mac[WIFI_MAC_ADDR_LEN];
	/** MAC address length */
	uint8_t mac_length;
	/** is TWT capable ? */
	bool twt_capable;
```

File: include/zephyr/net/wifi_mgmt.h [hunk 3]
```diff
@@ -802,6 +835,20 @@ void wifi_mgmt_raise_raw_scan_result_event(struct net_if *iface,
  */
 void wifi_mgmt_raise_disconnect_complete_event(struct net_if *iface, int status);
 
+/** Wi-Fi management AP mode enable result event
+ *
+ * @param iface Network interface
+ * @param status AP mode enable result status
+ */
+void wifi_mgmt_raise_ap_enable_result_event(struct net_if *iface, int status);
```
Comment (jukkar) [inline include/zephyr/net/wifi_mgmt.h:866]: Shouldn't the type of `status` be `enum wifi_ap_status`?

File: include/zephyr/net/wifi_mgmt.h [hunk 4]
```diff
@@ -802,6 +835,20 @@ void wifi_mgmt_raise_raw_scan_result_event(struct net_if *iface,
...
+
+/** Wi-Fi management AP mode disable result event
+ *
+ * @param iface Network interface
+ * @param status AP mode disable result status
+ */
+void wifi_mgmt_raise_ap_disable_result_event(struct net_if *iface, int status);
```
Comment (jukkar) [inline include/zephyr/net/wifi_mgmt.h:873]: here too

File: subsys/net/l2/wifi/wifi_shell.c [hunk 5]
```diff
@@ -64,6 +65,13 @@ static uint32_t scan_result;
 
 static struct net_mgmt_event_callback wifi_shell_mgmt_cb;
 
+K_MUTEX_DEFINE(wifi_ap_sta_list_lock);
```
Comment (jukkar) [inline subsys/net/l2/wifi/wifi_shell.c:68]: Could this be `static`?

File: subsys/net/l2/wifi/wifi_shell.c [hunk 6]
```diff
@@ -64,6 +65,13 @@ static uint32_t scan_result;
...
+struct wifi_ap_sta_node {
+	bool valid;
+	struct wifi_ap_sta_info sta_info;
+};
+struct wifi_ap_sta_node sta_list[CONFIG_WIFI_SHELL_MAX_AP_STA];
```
Comment (jukkar) [inline subsys/net/l2/wifi/wifi_shell.c:73]: perhaps `static`?

File: subsys/net/l2/wifi/wifi_shell.c [hunk 7]
```diff
@@ -328,17 +336,42 @@ static void handle_wifi_ap_disable_result(struct net_mgmt_event_callback *cb)
...
 	}
+
+	k_mutex_lock(&wifi_ap_sta_list_lock, K_FOREVER);
+	for (int i = 0; i < CONFIG_WIFI_SHELL_MAX_AP_STA; i++) {
+		if (!sta_list[i].valid) {
+			continue;
+		}
+
+		sta_list[i].valid = false;
+	}
```
Comment (jukkar) [inline subsys/net/l2/wifi/wifi_shell.c:347]: Could we just do `memset(&sta_list, 0, sizeof(sta_list));` here as the end result will be the same?

File: subsys/net/l2/wifi/wifi_shell.c [hunk 8]
```diff
@@ -1197,6 +1246,50 @@ static int cmd_wifi_ap_disable(const struct shell *sh, size_t argc,
 	return 0;
 }
 
+static int cmd_wifi_ap_stations(const struct shell *sh, size_t argc,
+				char *argv[])
+{
+	size_t id = 1;
+
+	(void)argv;
+	(void)argc;
```
Comment (jukkar) [inline subsys/net/l2/wifi/wifi_shell.c:1255]: Same result but as we have a macro for this, better to do
```
ARG_UNUSED(argv);
ARG_UNUSED(argc);
```

File: subsys/net/l2/wifi/wifi_shell.c [hunk 9]
```diff
@@ -1197,6 +1246,50 @@ static int cmd_wifi_ap_disable(const struct shell *sh, size_t argc,
...
+
+		if (!sta_list[i].valid) {
+			continue;
+		}
+
+		sta = &sta_list[i].sta_info;
+
+		shell_fprintf(sh, SHELL_NORMAL, "Station %zu:\n", id++);
+		shell_fprintf(sh, SHELL_NORMAL, "==========\n");
+		uint8_t mac_string_buf[sizeof("xx:xx:xx:xx:xx:xx")];
```
Comment (jukkar) [inline subsys/net/l2/wifi/wifi_shell.c:1272]: Please place variable declarations at the start of the block.

**Review by jukkar — COMMENTED** [review]
Overall looks good, couple of minor nits found.

**Review by jukkar — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**